### PR TITLE
adds the Project and Repo models

### PR DESCRIPTION
**Pull request by andrewschoen**

### CODE REVIEW

File: shaman/models/repos.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    __tablename__ = 'repos'
+    id = Column(Integer, primary_key=True)
+    url = Column(String(256))
+    chacra_url = Column(String(256))
+    ref = Column(String(256), index=True)
+    sha1 = Column(String(256), index=True)
+    distro = Column(String(256), nullable=False, index=True)
+    distro_version = Column(String(256), nullable=False, index=True)
+    modified = Column(DateTime, index=True)
+    signed = Column(Boolean(), default=False)
```
Comment (andrewschoen) [inline shaman/models/repos.py]: I'm not sure we should care about ``signed`` or ``type``. Thoughts?

File: shaman/models/projects.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    def __repr__(self):
+        try:
+            return '<Project %r>' % self.name
+        except DetachedInstanceError:
+            return '<Project detached>'
+
+    def __json__(self):
+        json_ = {}
+        json_["name"] = self.name
+        return json_
```
Comment (alfredodeza) [inline shaman/models/projects.py]: can't we just return the dict with `self.name` like:

    return dict(
        name=self.name,
    )

Why the back and forth?

Comment (andrewschoen) [inline shaman/models/projects.py]: No good reason, I'll update

File: shaman/models/projects.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+
+def get_or_create(name, **kw):
```
Comment (alfredodeza) [inline shaman/models/projects.py]: do we need `**kw` ? it is not passing it later

Comment (andrewschoen) [inline shaman/models/projects.py]: We don't, I think this was originally pulled from https://github.com/ceph/shaman/blob/master/shaman/models/__init__.py#L62 where ``**kw`` does make sense.

File: shaman/models/repos.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            distro_version=self.distro_version,
+            modified=self.modified,
+            signed=self.signed,
+            type=self.type,
+            status=self.status,
+        )
+
+
+# listen for timestamp modifications
+listen(Repo, 'before_insert', update_timestamp)
```
Comment (alfredodeza) [inline shaman/models/repos.py]: we are already getting timestamps from chacra, do we want to store a second timestamp local to shaman?

Comment (andrewschoen) [inline shaman/models/repos.py]: yeah, I think we care when the last time was shaman heard from chacra about a repo

File: shaman/models/repos.py (hunk 1 above)
Comment (alfredodeza) [inline shaman/models/repos.py]: we might want `signed` because these repos will need to be signed, `type` is not necessary - it is useful only in chacra

**Pull request merged**